yew/src/lib.rs

```diff
@@ -99,7 +99,119 @@ extern crate self as yew;
+///
+/// #[derive(Clone, Properties)]
+/// struct List {
+///   children: ChildrenRenderer<ListItem>,
+/// }
+/// impl Component for List {
+///   // ...
+/// #   type Message = ();
+/// #   type Properties = Self;
+/// #   fn create(props: Self::Properties, _: ComponentLink<Self>) -> Self { props }
+/// #   fn update(&mut self, _: Self::Message) -> ShouldRender { false }
+/// #   fn change(&mut self, _: Self::Properties) -> ShouldRender { false }
+/// #   fn view(&self) -> Html { unimplemented!() }
+/// }
+///
+/// #[derive(Clone)]
+/// struct ListItem;
+/// impl Component for ListItem {
+/// #   type Message = ();
+///   type Properties = ();
+///   // ...
+/// #   fn create(_: Self::Properties, _: ComponentLink<Self>) -> Self { Self }
+/// #   fn update(&mut self, _: Self::Message) -> ShouldRender { false }
+/// #   fn change(&mut self, _: Self::Properties) -> ShouldRender { false }
+/// #   fn view(&self) -> Html { unimplemented!() }
+/// }
+///
+/// // Required for ChildrenRenderer
+/// impl From<VChild<ListItem>> for ListItem {
+///   fn from(child: VChild<ListItem>) -> Self { Self }
+/// }
+///
+/// impl From<ListItem> for Html {
+///   fn from(item: ListItem) -> Html { item.view() }
+/// }
+///
+/// # fn test() -> Html {
+/// html! {
+///   <List>
+///     <ListItem/>
+///     <ListItem/>
+///     <ListItem/>
+///   </List>
+/// }
+/// # }
+/// ```
+///
+/// But in most real cases you need to forming content of you list programmatically.
+/// That's mean, that you would write code like this:
+/// ```
+/// # use yew::prelude::*;
+/// # use yew::html::ChildrenRenderer;
+/// # use yew::virtual_dom::VChild;
+/// # #[derive(Clone, Properties)]
+/// # struct List { children: ChildrenRenderer<ListItem> }
+/// # impl Component for List {
+/// #   type Message = ();
+/// #   type Properties = Self;
+/// #   fn create(props: Self::Properties, _: ComponentLink<Self>) -> Self { props }
+/// #   fn update(&mut self, _: Self::Message) -> ShouldRender { false }
+/// #   fn change(&mut self, _: Self::Properties) -> ShouldRender { false }
+/// #   fn view(&self) -> Html { unimplemented!() }
+/// # }
+/// # #[derive(Clone)]
+/// # struct ListItem;
+/// # impl Component for ListItem {
+/// #   type Message = ();
+/// #   type Properties = ();
+/// #   fn create(_: Self::Properties, _: ComponentLink<Self>) -> Self { Self }
+/// #   fn update(&mut self, _: Self::Message) -> ShouldRender { false }
+/// #   fn change(&mut self, _: Self::Properties) -> ShouldRender { false }
+/// #   fn view(&self) -> Html { unimplemented!() }
+/// # }
+/// # impl From<VChild<ListItem>> for ListItem {
+/// #   fn from(child: VChild<ListItem>) -> Self { Self }
+/// # }
+/// # impl From<ListItem> for Html {
+/// #   fn from(item: ListItem) -> Html { item.view() }
+/// # }
+/// # fn test() -> Html {
+/// # let some_iter = (0..10);
+/// html! {
+///   <List>
+///     { for some_iter.map(|_| html_nested!{ <ListItem/> }) }
+///   </List>
+/// }
+/// # }
```

Comment:
Can we move this into the previous code block to avoid repeating the setup code?
The lines above can be comments instead:
```rust
/// # fn test() -> Html {
/// # let some_iter = (0..10);
/// // In many cases you might want to create the content dynamically.
/// // To do this, you can use the following code:
/// html! {
///   <List>
///     { for some_iter.map(|_| html_nested! { <ListItem/> }) }
///   </List>
/// }
/// # }
```

We could also remove it entirely, I already added tests as part of #1531.

Reply:
Ok, will moved